### PR TITLE
[pipes] custom messages API docs

### PR DESCRIPTION
**Pull request by alangenfeld**
oversight not adding these in the initial PR

## How I Tested These Changes

view api docs

### CODE REVIEW

**Comment by alangenfeld** [comment]
Current dependencies on/for this PR:
* `master`
  * **PR #19172** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19172?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/19172?utm_source=stack-comment).

File: python_modules/dagster/dagster/_core/pipes/client.py [hunk 1]
```diff
@@ -111,6 +111,11 @@ def get_asset_check_result(self) -> AssetCheckResult:
         return _check_result_from_pipes_results(self.get_results())
 
     def get_custom_messages(self) -> Sequence[Any]:
+        """Get the sequence of deserialized from JSON data that was reported from the external process using
```
Comment (erinkcochran87) [inline python_modules/dagster/dagster/_core/pipes/client.py:114]: > deserialized from JSON data

"deserialized JSON data"? This is a bit difficult to understand - would this suggestion still be accurate? Or is there another way to make this clearer?

File: python_modules/dagster/dagster/_core/pipes/context.py [hunk 2]
```diff
@@ -344,12 +344,18 @@ def get_results(
...
-        Yields:
-            PipesExecutionResult: Result reported by external process.
+        Returns:
+            Sequence[PipesExecutionResult]: Result reported by external process.
         """
         return self.message_handler.get_reported_results()
 
+    @public
     def get_custom_messages(self) -> Sequence[Any]:
+        """Get the sequence of deserialized from JSON data that was reported from the external process using
```
Comment (erinkcochran87) [inline python_modules/dagster/dagster/_core/pipes/context.py:354]: Got one more: `deserialized JSON data`

Comment (alangenfeld) [reply python_modules/dagster/dagster/_core/pipes/context.py]: 🤦

**Review by erinkcochran87 — APPROVED** [review]
:shipit:

**Pull request merged**